pkg/imagehandler/imagefile.go

```diff
@@ -36,11 +36,26 @@ type imageFile struct {
 var _ fs.File = &imageFile{}
 
 func (f *imageFile) Init(inputFile baseFile) error {
-	if f.imageReader == nil {
-		var err error
-		ignition := &isoeditor.IgnitionContent{Config: f.ignitionContent}
-		f.imageReader, err = inputFile.InsertIgnition(ignition)
+	if f.imageReader != nil {
```

Comment:
Should we also check if f.size is set or non-zero? It is possible that there is a valid f.imageReader but we never got to set f.size correctly. 
If f.imageReader != nil
        if f.size == 0 { 
                attempt to calculate size again
        } else {
                 return nil
        }
}

Reply:
It's set at creation time: https://github.com/openshift/image-customization-controller/blob/main/pkg/imagehandler/imagehandler.go#L102

We should probably get rid of that, but for now let's get it working.